### PR TITLE
Add issue template for failing and flaky tests in testgrid

### PR DESCRIPTION
**Pull request by Jont828**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**: Copies the issue templates from CAPI on failing and flaky tests into CAPZ.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by Jont828** [comment]
@CecileRobertMichon

File: .github/ISSUE_TEMPLATE/flaking_test.md.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+name: Flaking Test
```
Comment (jackfrancis) [inline .github/ISSUE_TEMPLATE/flaking_test.md.md:2]: nit: change from "Flaking" to "Flaky"

**Comment by Jont828** [comment]
Are we good to approve/merge this?

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/assign @jackfrancis @dtzar

**Pull request merged**